### ISSUE TITLE
CA1861 false positive in method attribute

### ISSUE THREAD

**Issue by koenigst** [dotnet/roslyn-analyzers#7033]
### Analyzer

**Diagnostic ID**: [CA1861](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1861): `Avoid constant arrays as arguments`

### Analyzer source

**Version**: SDK 8.0.100

### Describe the bug

A constant array in an attribute initializer produces diagnostic CA1861 but extracting the array into a static field is not possible due to language constraints.

### Steps To Reproduce

1. Enable all analyzers.
```xml
    <EnableNETAnalyzers>true</EnableNETAnalyzers>
    <AnalysisMode>All</AnalysisMode>
```

2. Add attribute with array.
```csharp
public class Test
{
    [Xunit.Theory]
    [Xunit.InlineData("a", new[] { "b" })]
    public void Method(string a, string[] b) { }
}
```

**Expected Behavior**:
The array `new[] { "b" }` has no compiler warnings.

**Actual Behavior**:
The array `new[] { "b" }` has a CA1861 warning but the code cannot be changed because no field is allowed in attribute syntax.

**Comment by JeppeSN** [comment]
Happens for other attributes, like `NUnit.Framework.TestCaseAttribute` where it would be:
```csharp
[TestCase("a", new[] { "b" })]   // false CA1861
```
A hillarious workaround is to write:
```csharp
[Xunit.InlineData("a", (string[])["b"])]
```
where the cast `(string[])` is needed only if the attribute class constructor in question takes a very general argument, like `object`. If the receiver already wants a concrete array, `["b"]` without cast is sufficient.

**Comment by DoctorKrolic** [comment]
@mavasani How roslyn analyzers are updated for users? Are they bound to version of VS or .NET SDK or what? This issue if fixed sor quite some time, but I still have this bug (VS 17.9P5, latest .NET 8 SDK). Very annoying...
